frontend/src/tests/pages/RiderApplication/RiderApplicationIndexPageMember.test.js

```diff
@@ -0,0 +1,224 @@
+        mockConfirm.mockReturnValue(true);
+
+
+        const { getByTestId } = render(
+            <QueryClientProvider client={queryClient}>
+                <MemoryRouter>
+                    <RiderApplicationIndexPage />
+                </MemoryRouter>
+            </QueryClientProvider>
+        );
+
+        await waitFor(() => { expect(getByTestId(`${testId}-cell-row-2-col-id`)).toBeInTheDocument(); });
+
+        expect(getByTestId(`${testId}-cell-row-2-col-id`)).toHaveTextContent("4"); 
+
+
+        const tdCancelElement = document.querySelector('[data-testid="RiderApplicationTable-cell-row-2-col-Cancel"]');
+        const cancelButton = tdCancelElement.querySelector('button');
+        expect(cancelButton.classList.contains('btn-danger'));
+       
+        fireEvent.click(cancelButton);
+
+        expect(mockConfirm).toHaveBeenCalledWith("Are you sure you want to cancel this application?\n\nClick 'OK' to confirm or 'Cancel' to keep your application active.");
+        
+        // Use setTimeout to introduce a delay before checking the updated status
+        setTimeout(() => {
+            expect(getByTestId(`${testId}-cell-row-2-col-status`)).toHaveTextContent("cancelled");
+            }, 1000); 
+
+    });
+
+    test("what happens when you click cancel and no, member", async () => {
+        setupMemberOnly();
+
+        const queryClient = new QueryClient();
+        axiosMock.onGet("/api/rider").reply(200, riderApplicationFixtures.threeApplications);
+        axiosMock.onPut("/api/riderApplication/cancel").reply(200);
+
+        // Spy on window.confirm
+        const mockConfirm = jest.spyOn(window, "confirm");
+        mockConfirm.mockReturnValue(false);
+
+
+        const { getByTestId } = render(
+            <QueryClientProvider client={queryClient}>
+                <MemoryRouter>
+                    <RiderApplicationIndexPage />
+                </MemoryRouter>
+            </QueryClientProvider>
+        );
+
+        await waitFor(() => { expect(getByTestId(`${testId}-cell-row-2-col-id`)).toBeInTheDocument(); });
+
+        expect(getByTestId(`${testId}-cell-row-2-col-id`)).toHaveTextContent("4"); 
+
+
+        const tdCancelElement = document.querySelector('[data-testid="RiderApplicationTable-cell-row-2-col-Cancel"]');
+        const cancelButton = tdCancelElement.querySelector('button');
+        expect(cancelButton.classList.contains('btn-danger'));
+       
+        fireEvent.click(cancelButton);
+
+        expect(mockConfirm).toHaveBeenCalledWith("Are you sure you want to cancel this application?\n\nClick 'OK' to confirm or 'Cancel' to keep your application active.");
+        
+        // Use setTimeout to introduce a delay before checking the updated status
+        setTimeout(() => {
+        // Verify that the status is updated to "cancelled"
+            expect(getByTestId(`${testId}-cell-row-2-col-status`)).toHaveTextContent("pending");
+            }, 1000); // Adjust the delay duration as needed
+
```

Comment:
NO!  This is not how you do it.  This makes what we call a "flaky test", one that sometimes works and sometimes doesn't.    

Resist the temptation to solve race conditions in tests with timers.

Here's probably what It should be:

```suggestion

await waitFor( ()=> {
 expect(getByTestId(`${testId}-cell-row-2-col-status`)).toHaveTextContent("pending");
            }
```